ckanext/datastore/tests/test_chained_action.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+import nose
+
+import ckan.plugins as p
+import ckan.tests.helpers as helpers
+import ckan.tests.factories as factories
+
+assert_equals = nose.tools.assert_equals
+assert_raises = nose.tools.assert_raises
+
+
+deleted_count = 0
+
+
+@p.toolkit.chained_action
+def datastore_delete(up_func, context, data_dict):
+    global deleted_count
+    res = helpers.call_action(u"datastore_search",
+                              resource_id=data_dict[u'resource_id'],
+                              filters=data_dict[u'filters'],
+                              limit=10,)
+    result = up_func(context, data_dict)
+    deleted_count = res.get(u'total', 0)
+
+    return result
+
+
+class ChainedDataStorePlugin(p.SingletonPlugin):
```

Comment:
How about `ExampleDatastoreDeleteWithCountPlugin`

`ChainedDatastorePlugin` sounds like something a user might want to use..